pubsub/mqtt/mqtt.go

```diff
@@ -89,7 +89,7 @@ func (m *mqttPubSub) Init(metadata pubsub.Metadata) error {
 	m.ctx, m.cancel = context.WithCancel(context.Background())
 
 	// mqtt broker allows only one connection at a given time from a clientID.
-	producerClientID := fmt.Sprintf("%s-producer", m.metadata.clientID)
+	producerClientID := m.metadata.clientID
```

Comment:
This could be a breaking change. And why would you not also update the consumer with the same change?

If this PR is necessary (please explain in depth why) then instead I recommend introducing two new metadata options, `consumerClientID` and `producerClientID`. If those are set those values should be used, and if they are not set we should fall back to the current `clientID` metadata with the exact string we have today.

Reply:
Hello, sure . I will have a look. I am not really a golang expert hence might take some time. But please consider this change as critical. since current version is not a working version if i want to use a remote MQTT publisher.

For eg: I am subscribing to a remote mqtt producer who provides me with certificates and clientID(eg. 12345)

When I provide this as consumerID to Dapr , dapr sends request to the producer with clientID:12345-producer

and this is breaking functionality. Producer will never have "-producer" in the clientID only to suit the subscriber.

Currently Dapr MQTT is built to only support both producers and subscribers in sync with each other and but it's not the case of the time. So this change is critical else we are not able to move ahead with out Dapr+MQTT implementation. Request you to consider this as critical.

Reply:
With MQTT, we need two different client IDs for producer and consumer because we maintain 2 connections to the service. Setting them to the same value wouldn't work.

However, I understand the issue @mandarj123 is reporting. One way we could fix that is by introducing 2 new metadata options: 

- `producerClientID`
- `consumerClientID`

If those are set, we will use their values for the producer/consumer client IDs. When they're unset (one of them or both), we will use the current default ("%s-producer" or "%s-consumer").

Thoughts?

Reply:
I agree, and it would be great if someone can take this PR ahead and get this issue solved immediately.
My limited knowledge of golang means I could only do as many changes to remove the appending :)